framework/TSA/AutoCorrelation.py

```diff
@@ -0,0 +1,126 @@
+    """
+    specs = super(AutoCorrelation, cls).getInputSpecification()
+    specs.name = 'AutoCorrelation'
+    specs.description = """AutoCorrelation TimeSeriesAnalysis algorithm."""
+    specs.addSub(InputData.parameterInputFactory(
+      'nlags',
+      contentType=InputTypes.IntegerType,
+      descr=""" Number of lags to return autocorrelation for. """
+    ))
+    specs.addSub(InputData.parameterInputFactory(
+      'alpha',
+      contentType=InputTypes.FloatType,
+      descr=""" If a number is given, the confidence intervals for the given level are returned. For instance if alpha=.05, 
+      95 % confidence intervals are returned where the standard deviation is computed according to Bartlett”s formula. """
+    ))
+    return specs
+
+
+  def __init__(self, *args, **kwargs):
+    """
+      A constructor that will appropriately intialize a time-series analysis object
+      @ In, args, list, an arbitrary list of positional values
+      @ In, kwargs, dict, an arbitrary dictionary of keywords and values
+      @ Out, None
+    """
+    # general infrastructure
+    super().__init__(self, *args, **kwargs)
+
+
+  def handleInput(self, spec):
+    """
+      Reads user inputs into this object.
+      @ In, spec, InputData.InputParams, input specifications
+      @ Out, settings, dict, initialization settings for this algorithm
+    """
+    settings = super().handleInput(spec)
+    settings['nlags'] = spec.findFirst('nlags').value
+    settings['alpha'] = spec.findFirst('alpha').value
+    return settings
+
+
+  def characterize(self, signal, pivot, targets, settings):
+    """
+      This function utilizes the Discrete Wavelet Transform to
+      characterize a time-dependent series of data.
+
+      @ In, signal, np.ndarray, time series with dims [time, target]
+      @ In, pivot, np.1darray, time-like parameter values
+      @ In, targets, list(str), names of targets in same order as signal
+      @ In, settings, dict, additional settings specific to this algorithm
+      @ Out, params, dict, characteristic parameters
+    """
+    # TODO extend to continuous wavelet transform
+    try:
+      import statsmodels.tsa.stattools as sm
+    except ModuleNotFoundError:
+      print("This RAVEN TSA Module requires the statsmodels.tsa.stattools library to be installed in the current python environment")
+      raise ModuleNotFoundError
+
+
+    ## The pivot input parameter isn't used explicity in the
+    ## transformation as it assumed/required that each element in the
+    ## time-dependent series is independent, uniquely indexed and
+    ## sorted in time.
+    nlags = settings['nlags']
+    alpha = settings['alpha']
+    params = {target: {'results': {}} for target in targets}
+
+    for i, target in enumerate(targets):
+      results = params[target]['results']
+      results['signal1'], results['signal2'] = sm.acf(signal[:, i], nlags=nlags, alpha=alpha)
```

Comment:
I can't tell what `signal1` and `signal2` are by looking at them; can we name them something easier to decipher?

Reply:
I am not familiar with what we would be using for signal1 or signal2 in this case, Do you have suggestions for different names?

Reply:
I honestly don't know what they are. Maybe @dylanjm has some ideas.

Reply:
Dylan and I discussed it and we have changed signal1 to acf and signal2 to cofint.